src/main/java/io/vertx/httpproxy/impl/ParseUtils.java

```diff
@@ -10,26 +10,37 @@
  */
 package io.vertx.httpproxy.impl;
 
-import java.text.SimpleDateFormat;
-import java.time.DayOfWeek;
-import java.util.Date;
-import java.util.Locale;
-import java.util.TimeZone;
+import java.time.*;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.util.*;
 
 /**
  * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
  */
 public class ParseUtils {
 
-  public static Date parseHeaderDate(String value) {
+  public static final DateTimeFormatter RFC_850_DATE_TIME = new DateTimeFormatterBuilder()
+    .appendPattern("EEEE, dd-MMM-yy HH:mm:ss")
+    .parseLenient()
+    .appendLiteral(" UTC")
```

Comment:
```suggestion
    .appendLiteral(" GMT")
```
See https://www.rfc-editor.org/rfc/rfc9110.html#name-date-time-formats and https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Date